src/request_handler.rs

```diff
@@ -67,8 +68,8 @@ impl RequestHandler {
         }
         body {
           ul {
-            @while let Some(entry) = read_dir.next_entry().await? {
-              @let file_name = entry.file_name().to_string_lossy().into_owned();
+            @for entry in files {
+              @let file_name = entry.to_string_lossy().into_owned();
```

Comment:
What about `file_name{,s}`?

```suggestion
            @for file_name in file_names {
              @let file_name = file_name.to_string_lossy();
```

I think we can avoid calling `into_owned` because we're not creating a temporary.